nbparameterise/code.py

```diff
@@ -62,7 +71,15 @@ def extract_parameters(nb, lang=None):
     now, nbparameterise only handles 'python3' and 'python2'.
     """
     drv = get_driver_module(nb, override=lang)
-    params = list(drv.extract_definitions(first_code_cell(nb).source))
+    cell = findFirstParametersTaggedCell(nb)
+    if cell is None:
+        cell = first_code_cell(nb)
+        
+    if cell is None:
+        return {}
```

Comment:
Can we leave the behaviour with no code cells alone for now? We can discuss changing that separately if you think it's a good idea, but I think it's probably good for nbparameterise to error if it's given a notebook with no code cells.

Reply:
removed